Scripts/Source/minai_Sex.psc

```diff
@@ -1125,23 +970,17 @@ Event EndSexScene(int tid, bool HasPlayer)
     sslThreadController controller = slf.GetController(tid)
 
     Actor[] actorList = slf.HookActors(tid)
-    Actor[] targetactorList = actorList
-
+    
     ; Send event, AI can be aware SEX is happening here
     Actor[] sortedActorList = slf.SortActors(actorList,true)
     
-    ; Select an actor that's not the player, and have them talk.
-    actor otherActor = sortedActorList[0]
-    if otherActor == playerRef && sortedActorList.Length > 1
-      otherActor = sortedActorList[1]
-    EndIf
-
-    main.RegisterEvent(Main.GetActorName(sortedActorList[0])+ " and "+Main.GetActorName(sortedActorList[1])+ " ended the intimate moment","info_sexscene")
     if bHasAIFF
-      DirtyTalk(sortedActorList, "What did you think of the sex?")
-      AIFF.SetAnimationBusy(0, Main.GetActorName(otherActor))
+      ; TODO make weighted random selection of actor to speak
+      actor speaker = GetWeightedRandomActorToSpeak(sortedActorList)
+      sexTalkOnEnd(speaker)
+      AIFF.SetAnimationBusy(0, Main.GetActorName(speaker))
     EndIf
-    SetSexSceneState("off")
+    UpdateThreadTable("end", "sexlab", tid)
```

Comment:
Need to keep SetSexSceneState for tts integration

Reply:
answered in similar change above